server/routes/pool-management/delete-pool/delete-pool.controller.js

```diff
@@ -1,56 +1,71 @@
-      //  I am sure it will get cleaned once the user goes back to 'pool-management.get'
-      var pool = _.clone(req.session.poolDetails)
-        , confirmUrl = 'pool-management.delete-pool.post'
-        , cancelUrl
-        , members;
-
-      // if the user does not come from a pool-record we send them away
-      if (typeof pool === 'undefined') {
-        return res.redirect(app.namedRoutes.build('pool-management.get'));
+    return async function(req, res) {
+      const { poolNumber } = req.params;
+
+      let confirmUrl = 'pool-management.delete-pool.post';
+      let cancelUrl = '/pool-management/pool-overview/' + poolNumber;
+
+      let members, isActive;
+      try {
+        const response = await poolSummaryObj.get(req, poolNumber);
+        members = response.additionalStatistics.courtSupply + response.bureauSummoning.confirmed;
+        isActive = response.poolDetails.isActive;
+      } catch (err) {
+        app.logger.crit('Failed to fetch pool summary (delete pool journey): ', {
+          auth: req.session.authentication,
+          data: { poolNumber },
+          error: (typeof err.error !== 'undefined') ? err.error : err.toString(),
+        });
+
+        return res.render('_errors/generic');
       }
 
-      cancelUrl = '/pool-management/pool-overview/' + pool.poolDetails.poolNumber;
-      members = pool.additionalStatistics.courtSupply + pool.bureauSummoning.confirmed;
 
       if (members === 0) {
         return res.render('pool-management/delete-pool/confirm', {
-          poolNumber: pool.poolDetails.poolNumber,
+          poolNumber,
           confirmUrl: confirmUrl,
           cancelUrl: cancelUrl,
-          isActive: pool.poolDetails.isActive,
+          isActive,
         });
       }
 
       if (members > 0) {
         return res.render('pool-management/delete-pool/has-members', {
           cancelUrl: cancelUrl,
-          members: members,
-          isActive: pool.poolDetails.isActive,
+          members,
+          isActive,
         });
       }
     };
   };
 
   module.exports.postDeletePool = function(app) {
     return function(req, res) {
+      const { poolNumber } = req.params;
+
       var successCB = function() {
           req.session.deletedRecord = {
-            number: req.session.poolDetails.poolDetails.poolNumber,
+            number: poolNumber,
           };
 
+          app.logger.info('Successfully deleted pool: ', {
+            auth: req.session.authentication,
+            data: {
+              poolNumber,
+            },
+          });
+
           return res.redirect(app.namedRoutes.build('pool-management.get'));
         }
         , errorCB = function(err) {
-          var cancelUrl = '/pool-management/pool-overview/' + req.session.poolDetails.poolDetails.poolNumber;
+          var cancelUrl = '/pool-management/pool-overview/' + poolNumber;
 
           if (err.response.status === 423) {
```

Comment:
Getting an uncaught exception here when deleting a pool with jurors summoned.
Maybe update to;
`err.response?.status === 423`

Reply:
updated :D